includes/class-user-manual-sync.php

```diff
@@ -0,0 +1,239 @@
+		if ( ! class_exists( 'Newspack\Data_Events' ) ) {
+			return;
+		}
+
+		Data_Events::register_listener( 'newspack_network_manual_sync_user', 'network_manual_sync_user', [ __CLASS__, 'manual_sync_user' ] );
+	}
+
+	/**
+	 * Filters the user data for the event being triggered
+	 *
+	 * @param array $user_data The user data.
+	 * @return array
+	 */
+	public static function manual_sync_user( $user_data ) {
+		$synced_metadata = [];
+		$synced_props    = [];
+
+		// Create an array of all of the synced user meta values.
+		foreach ( self::$meta_to_sync as $key ) {
+			$synced_metadata[ $key ] = $user_data->$key;
+		}
+
+		foreach ( self::$user_props as $key ) {
+			$synced_props[ $key ] = $user_data->$key;
+		}
+
+		return [
+			'email'   => $user_data->user_email,
+			'role'    => array_shift( $user_data->roles ),
+			'user_id' => $user_data->ID,
+			'meta'    => $synced_metadata,
+			'prop'    => $synced_props,
+		];
+	}
+
+	/**
+	 * Create URL to manually sync user.
+	 *
+	 * @param WP_User $user The current WP_User object.
+	 * @return string $url The wp-admin URL.
+	 */
+	public static function get_manual_sync_link( $user ) {
+		$url = add_query_arg(
+			array(
+				'user_id'  => \esc_attr( $user->ID ),
+				'action'   => 'np_network_manual_user_sync',
+				'uid'      => \esc_attr( $user->ID ),
+				'_wpnonce' => \wp_create_nonce( 'np_network_manual_user_sync' ),
+			),
+			'/wp-admin/user-edit.php'
+		);
+
+		return $url;
+	}
+
+	/**
+	 * Add button to manually sync users to the Edit User Profile screen.
+	 *
+	 * @param WP_User $user The current WP_User object.
+	 */
+	public static function add_manual_sync_button( $user ) {
+		// TODO: should this option limited to specific roles?
+		if ( \current_user_can( 'edit_user', $user->ID ) ) :
+			?>
+		<div class="newspack-network-sync-user">
+			<h2><?php \esc_html_e( 'Newspack Network Tools', 'newspack-network' ); ?></h2>
+			<table class="form-table" role="presentation">
+				<tr>
+					<th>
+						<label><?php esc_html_e( 'Manually sync user', 'newspack-network' ); ?></label>
+					</th>
+					<td>
+						<a class="button" id="do-newspack-network-manual-sync" href="<?php echo esc_url( self::get_manual_sync_link( $user ) ); ?>">
+							<?php esc_html_e( 'Sync user across network', 'newspack-network' ); ?>
+						</a>
+						<p class="description">
+							<?php esc_html_e( 'Manually sync this user\'s information across all sites in your network, including their role. If this is a new user, clicking this button will also propigate the user account across your network.', 'newspack-network' ); ?>
+						</p>
+					</td>
+				</tr>
+			</table>
+		</div>
+			<?php
+	endif;
+	}
+
+	/**
+	 * Process the admin action request
+	 */
+	public static function process_admin_action() {
+		/** Allowed actions */
+		$actions = [
+			'np_network_manual_user_sync',
+		];
+
+		/** Add notice if admin action was successful. */
+		if ( isset( $_GET['update'] ) && in_array( $_GET['update'], $actions, true ) ) {
```

Comment:
I would also chose a more specific key here instead of `update`.

And a nitpck, since there is only one thing to check, no need to create an array and check in_array. You could simply do

` ... && $_GET['newspack_network_manual_sync_processed'] === 'success' `

Reply:
Done and done -- thank you! I got rid of the array entirely and just went with a single value 🙂